modules/ppcp-settings/docs/merchant-onboarding-flow.md

```diff
@@ -0,0 +1,263 @@
+   flags.canUseCasualSelling || step.id !== 'business'
+   ```
+
+2. **Payment Methods Step Filtering:**
+   ```javascript
+   // Skip payment methods screen if:
+   const isBrandedBCDC = ownBrandOnly && !flags.canUseCardPayments;
+   const shouldSkip = flags.shouldSkipPaymentMethods ||
+                     isCasualSeller ||
+                     isBrandedBCDC;
+   ```
+
+   **Conditions for skipping payment methods step:**
+   - `flags.shouldSkipPaymentMethods = true` (server-side business logic)
+   - Merchant selected casual seller account
+   - Store uses PayPal-only branding AND card payments unavailable
+
+## Component Visibility Control
+
+### Flag-Based Component Rendering
+
+Components throughout the onboarding flow check flags to determine visibility:
+
+**Example from Step Components:**
+```javascript
+// Business step - only renders if casual selling available
+const { flags } = OnboardingHooks.useFlags();
+if (!flags.canUseCasualSelling) {
+    return null; // Component not rendered
+}
+```
+
+### Screen Navigation Logic
+
+**File:** `modules/ppcp-settings/resources/js/Components/Screens/Onboarding/Steps/index.js`
+
+```javascript
+export const getCurrentStep = (requestedStep, steps) => {
+    // Ensures requested step exists in filtered steps array
+    const safeCurrentStep = isValidStep(requestedStep) ? requestedStep : 0;
+    return steps[safeCurrentStep];
+};
+```
+
+Navigation between steps validates:
+- Step index is within bounds of filtered steps
+- `canProceed()` function returns true for current step
+- Required data is collected before advancing
+
+### Initial State Hydration and State Persistence
+
+When the onboarding screen loads:
+
+1. **Frontend** makes GET request to `/wc/v3/wc_paypal/onboarding`
+2. **Backend** returns current state + flags via `OnboardingRestEndpoint::get_details()`
+3. **Frontend** initializes Redux store with server data
+4. **Components** render based on flags and current step
+
+User interactions trigger:
+
+1. **Frontend** updates local Redux state
+2. **Frontend** makes POST request to `/wc/v3/wc_paypal/onboarding` with changes
+3. **Backend** validates and persists via `OnboardingProfile::save()`
+4. **Backend** returns updated state (including any derived flag changes)
+
+## Key Integration Points
+
+### Gateway Synchronization
+
+**Files:**
+- `modules/ppcp-settings/src/Data/OnboardingProfile.php`
+- WordPress action: `woocommerce_paypal_payments_sync_gateways`
+
+```php
+public function set_gateways_synced(bool $synced): void {
+    $this->data['gateways_synced'] = $synced;
+    if ($synced) {
+        do_action('woocommerce_paypal_payments_sync_gateways');
+    }
+}
+```
+
+### Connection Flow Integration
+
+The onboarding integrates with PayPal's OAuth connection flow:
+- Final step provides connection buttons/forms
+- Connection success updates onboarding completion status
+- Failed connections allow retry without losing progress
+
```

Comment:
Suggestion: Add a link to the [authentication-flows.md](./authentication-flows.md) document